src/Android/Xamarin.Android/Samples/Layers/RasterRenderingRule/RasterRenderingRule.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            base.OnCreate(bundle);
+
+            Title = "Raster rendering rule";
+
+            // Create the UI, setup the control references and execute initialization 
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Assign a new map to the MapView
+            _myMapView.Map = new Map();
+
+            // Set the basemap to Streets
+            _myMapView.Map.Basemap = Basemap.CreateStreets();
+
+            // Set the Uri for the image server
+            _myUri = new Uri("https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer");
+
+            // Create a new image service raster from the Uri
+            ImageServiceRaster myImageServiceRaster = new ImageServiceRaster(_myUri);
+
+            // Load the image service raster
+            await myImageServiceRaster.LoadAsync();
+
+            // Get the ArcGIS image service info (aka. metadata) from the image service raster
+            ArcGISImageServiceInfo myArcGISImageServiceInfo = myImageServiceRaster.ServiceInfo;
+
+            // Get the full extent envelope of the image service raster (the Charlotte, NC area) 
+            Envelope myEnvelope = myArcGISImageServiceInfo.FullExtent;
+
+            // Define a new view point from the full extent envelope
+            Viewpoint myViewPoint = new Viewpoint(myEnvelope);
+
+            // Zoom to the area of the full extent envelope of the image service raster
+            await _myMapView.SetViewpointAsync(myViewPoint);
+
+            // Get the rendering rule info (i.e. definitions of how the image should be drawn) info from the image service raster
+            _myReadOnlyListRenderRuleInfos = myArcGISImageServiceInfo.RenderingRuleInfos;
+
+            // Make sure that we have at least one render rule info to proceed 
+            if (_myReadOnlyListRenderRuleInfos.Count > 0)
+            {
+                // Loop through each rendering rule info
+                foreach (RenderingRuleInfo myRenderingRuleInfo in _myReadOnlyListRenderRuleInfos)
+                {
+                    // Get the name of the rendering rule info
+                    string myRenderingRuleName = myRenderingRuleInfo.Name;
+
+                    // Add the name of the rendering rule info into the global list of names
+                    _names.Add(myRenderingRuleName);
+                }
+            }
+
+            // Invoke the button for the user to change the rendering rule of the image service raster
+            OnChangeRenderingRuleClicked(_renderingRulesButton, null);
+        }
+
+        private void OnChangeRenderingRuleClicked(object sender, EventArgs e)
+        {
+            // Get the rendering rule button
+            var renderingRuleButton = sender as Button;
+
+            // Create menu to show the rendering rule options
+            var renderingRuleMenu = new PopupMenu(this, renderingRuleButton);
```

Comment:
I don't like the use of `var` here and throughout the sample. Although var is very convenient, it doesn't read well in code listings outside of an IDE.